pydal/objects.py

```diff
@@ -2122,7 +2269,7 @@ def validate_and_update(self, **update_fields):
         return response
 
     def delete_uploaded_files(self, upload_fields=None):
-        table = self.db[self.db._adapter.tables(self.query)[0]]
+        table = self.db._adapter.tables(self.query).popitem()[1]
```

Comment:
The new code is functionally equivalent to the original but neither version will work properly if there's more than 1 table in the query.